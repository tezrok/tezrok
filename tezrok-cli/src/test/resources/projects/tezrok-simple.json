{
  "name": "simple-app",
  "version": "0.0.1",
  "package": "com.tezrok.simple",
  "modules": [
    {
      "name": "simpleModule",
      "type": "tezrok",
      "importSchema": "./schemas/Orders.json"
    }
  ]
}
