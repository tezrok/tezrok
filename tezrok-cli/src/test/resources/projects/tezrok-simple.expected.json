{
  "name": "simple-app",
  "version": "0.0.1",
  "description": "",
  "package": "com.tezrok.simple",
  "author": "",
  "modules": [
    {
      "name": "simpleModule",
      "description": "",
      "type": "tezrok",
      "schema": {
        "schemaName": "smpl",
        "importSchema": "./schemas/Orders.json",
        "entities": [
          {
            "name": "Order",
            "customRepository": true,
            "fields": [
              {
                "name": "id",
                "type": "Long",
                "required": true,
                "primary": true
              },
              {
                "name": "name",
                "type": "String",
                "required": true,
                "unique": true
              },
              {
                "name": "category",
                "type": "String",
                "required": true
              },
              {
                "name": "qty",
                "type": "Integer",
                "required": false
              },
              {
                "name": "created",
                "type": "DateTime",
                "required": false
              },
              {
                "name": "selectedItem",
                "type": "Item",
                "description": "The item that was selected by the user",
                "required": false,
                "logicField": true,
                "relation": "ManyToOne"
              },
              {
                "name": "selectedItemId",
                "type": "Long",
                "foreignField": "Item.id",
                "description": "Synthetic field for \"Order.selectedItem\"",
                "required": false,
                "syntheticTo": "Order.selectedItem"
              },
              {
                "name": "items",
                "type": "Item",
                "required": false,
                "logicField": true,
                "relation": "OneToMany"
              },
              {
                "name": "otherItems",
                "type": "Item",
                "required": false,
                "logicField": true,
                "relation": "ManyToMany"
              },
              {
                "name": "active",
                "type": "Boolean",
                "required": true
              },
              {
                "name": "nextOrder",
                "type": "Order",
                "logicField": true,
                "relation": "OneToOne"
              },
              {
                "name": "nextOrderId",
                "type": "Long",
                "foreignField": "Order.id",
                "description": "Synthetic field for \"Order.nextOrder\"",
                "unique": true,
                "syntheticTo": "Order.nextOrder"
              },
              {
                "name": "linkedOrders",
                "type": "Order",
                "logicField": true,
                "relation": "ManyToMany"
              }
            ],
            "customMethods": [
              "getByName",
              "getByNextOrderId",
              "findOrderLinkedOrdersByOrderId",
              "findOrderLinkedOrdersByOrderIdIn",
              "findIdSelectedItemIdNextOrderIdByIdIn",
              "findItemRelatedOrdersByItemId",
              "findItemRelatedOrdersByItemIdIn"
            ],
            "customComments": {
              "findIdSelectedItemIdNextOrderIdByIdIn": "Returns specified fields (id, selectedItemId, nextOrderId) of {@link OrderDto} into custom class.",
              "findItemRelatedOrdersByItemId": "Returns list of {@link OrderDto} to support ManyToMany relation for field {@link ItemFullDto#relatedOrders}.",
              "findItemRelatedOrdersByItemIdIn": "Returns list of primary field of {@link OrderDto} to support ManyToMany relation for field {@link ItemFullDto#relatedOrders}.",
              "findOrderLinkedOrdersByOrderId": "Returns list of {@link OrderDto} to support ManyToMany relation for field {@link OrderFullDto#linkedOrders}.",
              "findOrderLinkedOrdersByOrderIdIn": "Returns list of primary field of {@link OrderDto} to support ManyToMany relation for field {@link OrderFullDto#linkedOrders}.",
              "getByName": "Returns {@link OrderDto} by unique field {@link OrderDto#name}.",
              "getByNextOrderId": "Returns {@link OrderDto} by unique field {@link OrderDto#nextOrderId}."
            }
          },
          {
            "name": "Item",
            "customRepository": true,
            "activable": true,
            "createdAt": true,
            "updatedAt": true,
            "fields": [
              {
                "name": "id",
                "type": "Long",
                "required": true,
                "primary": true
              },
              {
                "name": "title",
                "type": "String",
                "required": false
              },
              {
                "name": "price",
                "type": "Double",
                "required": true
              },
              {
                "name": "otherItems",
                "type": "Item",
                "logicField": true,
                "relation": "ManyToMany"
              },
              {
                "name": "relatedOrders",
                "type": "Order",
                "description": "Related by this item orders",
                "logicField": true,
                "relation": "ManyToMany"
              },
              {
                "name": "itemsOrderId",
                "type": "Long",
                "foreignField": "Order.id",
                "description": "Synthetic field for \"Order.items\"",
                "required": false,
                "syntheticTo": "Order.items"
              },
              {
                "name": "active",
                "type": "Boolean",
                "description": "Is entity active or deleted",
                "required": true,
                "defValue": "false"
              },
              {
                "name": "createdAt",
                "type": "DateTime",
                "description": "Date and time of entity creation",
                "required": true,
                "defValue": "now()"
              },
              {
                "name": "updatedAt",
                "type": "DateTime",
                "description": "Date and time of entity update",
                "required": true,
                "defValue": "now()"
              }
            ],
            "customMethods": [
              "findOrderOtherItemsByOrderIdAndItemTitle",
              "findOrderItemsByItemItemsOrderIdAndItemTitle",
              "findOrderOtherItemsByOrderId",
              "findOrderOtherItemsByOrderIdIn",
              "findOrderItemsByItemItemsOrderId",
              "findIdByItemsOrderId",
              "findIdItemsOrderIdByItemsOrderIdIn",
              "findItemOtherItemsByItemId",
              "findItemOtherItemsByItemIdIn",
              "findIdItemsOrderIdByIdIn"
            ],
            "customComments": {
              "findIdByItemsOrderId": "Returns list of primary field of {@link ItemDto} to support OneToMany relation for field {@link OrderFullDto#items}.",
              "findIdItemsOrderIdByIdIn": "Returns specified fields (id, itemsOrderId) of {@link ItemDto} into custom class.",
              "findIdItemsOrderIdByItemsOrderIdIn": "Returns specified fields (id, itemsOrderId) of {@link ItemDto} into custom class to support OneToMany relation for field {@link OrderFullDto#items}",
              "findItemOtherItemsByItemId": "Returns list of {@link ItemDto} to support ManyToMany relation for field {@link ItemFullDto#otherItems}.",
              "findItemOtherItemsByItemIdIn": "Returns list of primary field of {@link ItemDto} to support ManyToMany relation for field {@link ItemFullDto#otherItems}.",
              "findOrderItemsByItemItemsOrderId": "Returns list of {@link ItemDto} to support OneToMany relation for field {@link OrderFullDto#items}.",
              "findOrderOtherItemsByOrderId": "Returns list of {@link ItemDto} to support ManyToMany relation for field {@link OrderFullDto#otherItems}.",
              "findOrderOtherItemsByOrderIdIn": "Returns list of primary field of {@link ItemDto} to support ManyToMany relation for field {@link OrderFullDto#otherItems}."
            }
          },
          {
            "name": "Info",
            "fields": [
              {
                "name": "infoId",
                "type": "Long",
                "description": "Entity with \"infoId\" as \"id\" (not default)",
                "primary": true
              },
              {
                "name": "name",
                "type": "String",
                "maxLength": 100
              }
            ]
          },
          {
            "name": "User",
            "description": "User entity",
            "customRepository": true,
            "createdAt": true,
            "updatedAt": true,
            "fields": [
              {
                "name": "id",
                "type": "Long",
                "primary": true
              },
              {
                "name": "name",
                "type": "String",
                "required": true,
                "minLength": 3,
                "maxLength": 20,
                "unique": true
              },
              {
                "name": "email",
                "type": "String",
                "required": true,
                "minLength": 4,
                "maxLength": 40,
                "unique": true
              },
              {
                "name": "password",
                "type": "String",
                "required": true,
                "maxLength": 100
              },
              {
                "name": "activated",
                "type": "Boolean",
                "required": true,
                "defValue": "false"
              },
              {
                "name": "banned",
                "type": "Boolean",
                "required": true,
                "defValue": "false"
              },
              {
                "name": "roles",
                "type": "Role",
                "logicField": true,
                "relation": "ManyToMany"
              },
              {
                "name": "createdAt",
                "type": "DateTime",
                "description": "Date and time of entity creation",
                "required": true,
                "defValue": "now()"
              },
              {
                "name": "updatedAt",
                "type": "DateTime",
                "description": "Date and time of entity update",
                "required": true,
                "defValue": "now()"
              }
            ],
            "customMethods": [
              "getByNameOrEmail",
              "getByName",
              "getByEmail"
            ],
            "customComments": {
              "getByEmail": "Returns {@link UserDto} by unique field {@link UserDto#email}.",
              "getByName": "Returns {@link UserDto} by unique field {@link UserDto#name}.",
              "getByNameOrEmail": "Returns {@link UserDto} by name or email."
            },
            "init": "admin,admin@site.com,admin"
          },
          {
            "name": "Role",
            "description": "Role entity",
            "customRepository": true,
            "createdAt": true,
            "updatedAt": true,
            "fields": [
              {
                "name": "id",
                "type": "Long",
                "primary": true
              },
              {
                "name": "name",
                "type": "String",
                "required": true,
                "minLength": 3,
                "maxLength": 20,
                "unique": true
              },
              {
                "name": "description",
                "type": "String",
                "minLength": 0,
                "maxLength": 150
              },
              {
                "name": "permissions",
                "type": "Permission",
                "logicField": true,
                "relation": "ManyToMany"
              },
              {
                "name": "createdAt",
                "type": "DateTime",
                "description": "Date and time of entity creation",
                "required": true,
                "defValue": "now()"
              },
              {
                "name": "updatedAt",
                "type": "DateTime",
                "description": "Date and time of entity update",
                "required": true,
                "defValue": "now()"
              }
            ],
            "customMethods": [
              "getByName",
              "findUserRolesByUserId",
              "findUserRolesByUserIdIn"
            ],
            "customComments": {
              "findUserRolesByUserId": "Returns list of {@link RoleDto} to support ManyToMany relation for field {@link UserFullDto#roles}.",
              "findUserRolesByUserIdIn": "Returns list of primary field of {@link RoleDto} to support ManyToMany relation for field {@link UserFullDto#roles}.",
              "getByName": "Returns {@link RoleDto} by unique field {@link RoleDto#name}."
            },
            "init": "ADMIN,Administrator role\nUSER,Authenticated role"
          },
          {
            "name": "Permission",
            "description": "Permission entity",
            "customRepository": true,
            "createdAt": true,
            "updatedAt": true,
            "fields": [
              {
                "name": "id",
                "type": "Long",
                "primary": true
              },
              {
                "name": "name",
                "type": "String",
                "required": true,
                "minLength": 3,
                "maxLength": 20,
                "unique": true
              },
              {
                "name": "description",
                "type": "String",
                "minLength": 0,
                "maxLength": 150
              },
              {
                "name": "createdAt",
                "type": "DateTime",
                "description": "Date and time of entity creation",
                "required": true,
                "defValue": "now()"
              },
              {
                "name": "updatedAt",
                "type": "DateTime",
                "description": "Date and time of entity update",
                "required": true,
                "defValue": "now()"
              }
            ],
            "customMethods": [
              "getByName",
              "findRolePermissionsByRoleId",
              "findRolePermissionsByRoleIdIn"
            ],
            "customComments": {
              "findRolePermissionsByRoleId": "Returns list of {@link PermissionDto} to support ManyToMany relation for field {@link RoleFullDto#permissions}.",
              "findRolePermissionsByRoleIdIn": "Returns list of primary field of {@link PermissionDto} to support ManyToMany relation for field {@link RoleFullDto#permissions}.",
              "getByName": "Returns {@link PermissionDto} by unique field {@link PermissionDto#name}."
            }
          },
          {
            "name": "OrderItemOtherItems",
            "description": "Synthetic entity of many-to-many relation for field \"Order.otherItems\"",
            "syntheticTo": "Order.otherItems",
            "fields": [
              {
                "name": "orderId",
                "type": "Long",
                "foreignField": "Order.id",
                "primary": true,
                "syntheticTo": "Order.otherItems"
              },
              {
                "name": "itemId",
                "type": "Long",
                "foreignField": "Item.id",
                "primary": true,
                "syntheticTo": "Order.otherItems"
              }
            ],
            "customMethods": [
              "findByOrderId",
              "findByItemId",
              "findByOrderIdIn",
              "findByItemIdIn"
            ]
          },
          {
            "name": "OrderOrderLinkedOrders",
            "description": "Synthetic entity of many-to-many relation for field \"Order.linkedOrders\"",
            "syntheticTo": "Order.linkedOrders",
            "fields": [
              {
                "name": "orderSourceId",
                "type": "Long",
                "foreignField": "Order.id",
                "primary": true,
                "syntheticTo": "Order.linkedOrders"
              },
              {
                "name": "orderTargetId",
                "type": "Long",
                "foreignField": "Order.id",
                "primary": true,
                "syntheticTo": "Order.linkedOrders"
              }
            ],
            "customMethods": [
              "findByOrderSourceId",
              "findByOrderTargetId",
              "findByOrderSourceIdIn",
              "findByOrderTargetIdIn"
            ]
          },
          {
            "name": "ItemItemOtherItems",
            "description": "Synthetic entity of many-to-many relation for field \"Item.otherItems\"",
            "syntheticTo": "Item.otherItems",
            "fields": [
              {
                "name": "itemSourceId",
                "type": "Long",
                "foreignField": "Item.id",
                "primary": true,
                "syntheticTo": "Item.otherItems"
              },
              {
                "name": "itemTargetId",
                "type": "Long",
                "foreignField": "Item.id",
                "primary": true,
                "syntheticTo": "Item.otherItems"
              }
            ],
            "customMethods": [
              "findByItemSourceId",
              "findByItemTargetId",
              "findByItemSourceIdIn",
              "findByItemTargetIdIn"
            ]
          },
          {
            "name": "ItemOrderRelatedOrders",
            "description": "Synthetic entity of many-to-many relation for field \"Item.relatedOrders\"",
            "syntheticTo": "Item.relatedOrders",
            "fields": [
              {
                "name": "itemId",
                "type": "Long",
                "foreignField": "Item.id",
                "primary": true,
                "syntheticTo": "Item.relatedOrders"
              },
              {
                "name": "orderId",
                "type": "Long",
                "foreignField": "Order.id",
                "primary": true,
                "syntheticTo": "Item.relatedOrders"
              }
            ],
            "customMethods": [
              "findByItemId",
              "findByOrderId",
              "findByItemIdIn",
              "findByOrderIdIn"
            ]
          },
          {
            "name": "UserRoleRoles",
            "description": "Synthetic entity of many-to-many relation for field \"User.roles\"",
            "syntheticTo": "User.roles",
            "fields": [
              {
                "name": "userId",
                "type": "Long",
                "foreignField": "User.id",
                "primary": true,
                "syntheticTo": "User.roles"
              },
              {
                "name": "roleId",
                "type": "Long",
                "foreignField": "Role.id",
                "primary": true,
                "syntheticTo": "User.roles"
              }
            ],
            "customMethods": [
              "findByUserId",
              "findByRoleId",
              "findByUserIdIn",
              "findByRoleIdIn"
            ]
          },
          {
            "name": "RolePermissionPermissions",
            "description": "Synthetic entity of many-to-many relation for field \"Role.permissions\"",
            "syntheticTo": "Role.permissions",
            "fields": [
              {
                "name": "roleId",
                "type": "Long",
                "foreignField": "Role.id",
                "primary": true,
                "syntheticTo": "Role.permissions"
              },
              {
                "name": "permissionId",
                "type": "Long",
                "foreignField": "Permission.id",
                "primary": true,
                "syntheticTo": "Role.permissions"
              }
            ],
            "customMethods": [
              "findByRoleId",
              "findByPermissionId",
              "findByRoleIdIn",
              "findByPermissionIdIn"
            ]
          }
        ],
        "enums": []
      },
      "auth": {
        "type": "simple",
        "stdInit": true
      },
      "dependencies": [
        "commons-io:commons-io:2.13.0"
      ],
      "properties": {}
    }
  ]
}